File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/scenemanagement/internal/ZComparator.java
package com.abubusoft.xenon.scenemanagement.internal;

import java.util.Comparator;

import com.abubusoft.xenon.entity.BaseEntity;

public class ZComparator<E extends BaseEntity> implements Comparator<E> {

	/* (non-Javadoc)
	 * @see java.util.Comparator#compare(java.lang.Object, java.lang.Object)
	 */
	@Override
	public int compare(E entity1, E entity2) {
		return (int)(entity1.position.z - entity2.position.z);
	}

}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/scenemanagement/internal/ZComparator.kt
package com.abubusoft.xenon.scenemanagement.internal

import com.abubusoft.xenon.entity.BaseEntity

class ZComparator<E : BaseEntity?> : Comparator<E> {
    /* (non-Javadoc)
	 * @see java.util.Comparator#compare(java.lang.Object, java.lang.Object)
	 */
    override fun compare(entity1: E, entity2: E): Int {
        return (entity1!!.position.z - entity2!!.position.z).toInt()
    }
}